src/utils/useBuildEvents.ts

```diff
@@ -15,18 +15,32 @@ export const useBuildEvents = ({
   const emit = useChannel({});
   const trackEvent = useContext(TelemetryContext);
   const [isStarting, setStarting] = useState(false);
+  const [isDisallowed, setDisallowed] = useState(false);
+
+  const isCancelable = localBuildProgress
+    ? ["initialize", "build", "upload"].includes(localBuildProgress?.currentStep)
+    : false;
+
+  const isRunning = localBuildProgress
+    ? !["aborted", "complete", "error", "limited"].includes(localBuildProgress.currentStep)
+    : isStarting;
 
   const startBuild = useCallback(() => {
+    setDisallowed(false);
     setStarting(true);
     emit(START_BUILD, { accessToken });
     trackEvent?.({ action: "startBuild" });
   }, [accessToken, emit, trackEvent]);
 
   const stopBuild = useCallback(() => {
-    setStarting(false);
-    emit(STOP_BUILD);
-    trackEvent?.({ action: "stopBuild" });
-  }, [emit, trackEvent]);
+    if (!isCancelable) {
+      setDisallowed(true);
```

Comment:
Shouldn't the function just throw an error instead? I don't see considerable advantages in introducing a side effect by setting a state here. This adds unnecessary decoupling. But it's up to you. Not a blocker for me.

Reply:
We need the state value to wobble the button when clicking it after the upload step. Throwing or doing nothing would be unexpected behavior missing user feedback.